bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py

```diff
@@ -0,0 +1,250 @@
+
+def _PyFilter(rdf, func_or_str, *args , extra_args = {} ):
+    """
+    Filters the entries of RDF according to a given condition.
+    Arguments:
+        func_or_str: The condition according to which the RDF is to be filtered. 
+            It can be either a python callable or a c style string.
+        *args:
+            Can be at most two of them.
+            a. The list of columns in  which case the starting arguments of the provided function callable in mapped to the columns in the list.
+                If not provide then it tries maps the name of the parameter to a column name of the RDF. Flags an error if not found.
+            b. The name of the Filter as a string.
+        extra_args: If the function needs any other parameters that are not columns and must be passed explicitly.
+    Returns:
+        self: The filtered instance of the RDF.
+    
+    Examples:
+    1. rdf.Filter("x>y")
+        Filter using an c-style expression
+    2. rdf.Filter(lambda x,y: x>y)
+        Filter using a python callable.
+    3. rdf.Filter(lambda x,y: x>y, ["y", "x"])
+        Filter using a python callable with column list passed explicitly.
+        This is equivalent to the filter "y>x".
+    """
+    if isinstance(func_or_str, str): # If string argument is passed. Invoke the Original Filters.
+        return rdf._OriginalFilter(func_or_str, *args)
+
+    if not callable(func_or_str): # The 1st argument is either a string or a python callable.
+        raise ValueError("Filter takes in either a string or a python callable as the first argument.")
+    
+    if len(args) > 2: 
+        raise ValueError("Filter takes in at most 4 arguments (func, cols, filter_name, extra_args).")
+    
+    func = func_or_str
+    jitter = FunctionJitter(rdf)
+    func.__annotations__['return'] = 'bool' # return type for Filters is bool # Note: You can keep double and Filter still works.
+
+    col_list = []
+    filter_name  = ""
+
+    if len(args) == 0:
+        pass
+    
+    if len(args) == 1:
+        if isinstance(args[0], list): 
+            col_list = args[0]
+        elif isinstance(args[0], str):
+            filter_name = args[0]
+        else:
+            raise ValueError("Invalid input obtained.") # ! TODO: Make this error better
+    
+    if len(args) == 2:
+        if isinstance(args[0], list) and isinstance(args[1], str):
+            col_list = args[0]; filter_name = args[1]
+        elif isinstance(args[1], list) and isinstance(args[0], str):
+            col_list = args[1]; filter_name = args[0]
+        else:
+            raise ValueError("Invalid input obtained.") # ! TODO: Make this error better
+    
+    func_call = jitter.jit_function(func, col_list, extra_args)
+    return rdf._OriginalFilter("Numba::" + func_call, filter_name)
+
+
+# This class is still to undergo some restructuring thus the docstring may not be complete.
+# The class will be fully realized after the Define has been Pythonized.
+class FunctionJitter:
```

Comment:
The class should be defined before the `_PyFilter`, since it is used by it.

Reply:
Done. But won't it affect readability?

Reply:
Yes, I believe it improves readability